From 2df11de9726713bfbe2f612f5fe95d89ac29f8e7 Mon Sep 17 00:00:00 2001
From: Jenny TC <jenny.tc@intel.com>
Date: Fri, 20 Feb 2015 08:23:51 +0530
Subject: [PATCH] psy_charger: Change Type C cable order

On disconnecting charger the status is displayed as Not Charging.
This is fixed by moving the Type C cable list as a second last entry
in the cable list to reflect the cable status properly.

Change-Id: If7c9bdb1d535b0a8702ede96f4840a638eccac23
Signed-off-by: Jenny TC <jenny.tc@intel.com>
---
 drivers/power/power_supply_charger.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/drivers/power/power_supply_charger.c b/drivers/power/power_supply_charger.c
index c4c3cdc..b2b66e3 100644
--- a/drivers/power/power_supply_charger.c
+++ b/drivers/power/power_supply_charger.c
@@ -67,13 +67,13 @@ static struct charger_cable cable_list[] = {
 	 .extcon_cable_type = EXTCON_TA,
 	 },
 	{
-	 .psy_cable_type = POWER_SUPPLY_CHARGER_TYPE_AC,
-	 .extcon_cable_type = EXTCON_AC,
-	 },
-	{
 	 .psy_cable_type = POWER_SUPPLY_CHARGER_TYPE_USB_TYPEC,
 	 .extcon_cable_type = EXTCON_TYPEC,
 	},
+	{
+	 .psy_cable_type = POWER_SUPPLY_CHARGER_TYPE_AC,
+	 .extcon_cable_type = EXTCON_AC,
+	 },
 };
 
 static int get_supplied_by_list(struct power_supply *psy,
@@ -105,12 +105,12 @@ struct charger_cable *get_cable(unsigned long usb_chrgr_type)
 		return &cable_list[3];
 	case POWER_SUPPLY_CHARGER_TYPE_ACA_DOCK:
 		return &cable_list[4];
-	case POWER_SUPPLY_CHARGER_TYPE_AC:
-		return &cable_list[6];
-	case POWER_SUPPLY_CHARGER_TYPE_USB_TYPEC:
-		return &cable_list[7];
 	case POWER_SUPPLY_CHARGER_TYPE_SE1:
 		return &cable_list[5];
+	case POWER_SUPPLY_CHARGER_TYPE_USB_TYPEC:
+		return &cable_list[6];
+	case POWER_SUPPLY_CHARGER_TYPE_AC:
+		return &cable_list[7];
 	}
 
 	return NULL;
-- 
1.9.1

